Nos/Views/NoteButton.swift

```diff
@@ -7,42 +7,90 @@
 
 import Foundation
 import SwiftUI
+import CoreData
 
 /// This view displays the a button with the information we have for a note suitable for being used in a list
 /// or grid.
 ///
 /// The button opens the ThreadView for the note when tapped.
 struct NoteButton: View {
-    var note: Event
+
+    @State var note: Event?
+    var noteID: HexadecimalString
     var style = CardStyle.compact
     var showFullMessage = false
     var hideOutOfNetwork = true
     var allowsPush = true
     var showReplyCount = true
     var isInThreadView = false
 
+    @Environment(\.managedObjectContext) private var viewContext
     @EnvironmentObject private var router: Router
     @EnvironmentObject private var relayService: RelayService
 
+    init(
+        note: Event, 
+        style: CardStyle = CardStyle.compact, 
+        showFullMessage: Bool = false, 
+        hideOutOfNetwork: Bool = true, 
+        allowsPush: Bool = true, 
+        showReplyCount: Bool = true, 
+        isInThreadView: Bool = false
+    ) {
+        self.note = note
+        self.noteID = note.identifier!
+        self.style = style
+        self.showFullMessage = showFullMessage
+        self.hideOutOfNetwork = hideOutOfNetwork
+        self.allowsPush = allowsPush
+        self.showReplyCount = showReplyCount
+        self.isInThreadView = isInThreadView
+    }
+
+    init(
+        noteID: HexadecimalString, 
+        style: CardStyle = CardStyle.compact, 
+        showFullMessage: Bool = false, 
+        hideOutOfNetwork: Bool = true, 
+        allowsPush: Bool = true, 
+        showReplyCount: Bool = true, 
+        isInThreadView: Bool = false
+    ) {
+        self.noteID = noteID
+        self.style = style
+        self.showFullMessage = showFullMessage
+        self.hideOutOfNetwork = hideOutOfNetwork
+        self.allowsPush = allowsPush
+        self.showReplyCount = showReplyCount
+        self.isInThreadView = isInThreadView
+    }
+
     var body: some View {
-        Button {
-            if allowsPush {
-                if !isInThreadView, let referencedNote = note.referencedNote() {
-                    router.push(referencedNote)
-                } else {
-                    router.push(note)
+        VStack {
```

Comment:
I use Group for these cases where the sole need is to join a conditional. https://developer.apple.com/documentation/swiftui/group uses it in a analogous example.

Reply:
Ok, I'll use that from now on.